src/__tests__/deprecation.test.ts

```diff
@@ -1,30 +1,46 @@
-import { IsString, applyTransformer, withDeprecationWarnings } from '..';
+import {
+    IsString as DefaultIsString,
+    applyTransformer,
+    withDeprecationWarnings,
+} from '..';
 
 describe('warnIfDeprecated', () => {
     it('produces new decorator factories', () => {
-        const onWarning = jest.fn();
+        const spy = jest.spyOn(process, 'emitWarning');
 
-        const IsStringWithDeprecationWarning = applyTransformer(
-            IsString,
-            withDeprecationWarnings(onWarning),
+        const IsString = applyTransformer(
+            DefaultIsString,
+            withDeprecationWarnings,
         );
 
         class Example {
-            @IsStringWithDeprecationWarning({
+            @IsString({
                 deprecated: true,
             })
             foo!: string;
 
-            @IsStringWithDeprecationWarning({
+            @IsString({
             })
             bar!: string;
+
+            @IsString({
+                deprecated: true,
+            })
+            baz!: string;
         }
 
         const example = new Example();
         example.foo = 'warning expected';
+        example.foo = 'no warning expected';
```

Comment:
Note that:
 - I'm setting `foo` twice; we expect only one warning.
 - I'm setting `baz` as well; we expect a warning here (even if `foo` has been silenced)